src/components/BeerDetails.js

```diff
@@ -0,0 +1,46 @@
+import  axios  from 'axios';
+import { useEffect, useState } from 'react';
+import { useParams  } from 'react-router-dom';
+
+function BeerDetails() {
+  
+    const {id} = useParams();
+
+    const [details, setDetails] = useState(null)
+
+    const baseUrl = 'https://ih-beers-api2.herokuapp.com/beers';
+       
+    useEffect(() => {
+      axios
+        .get(baseUrl + id)
```

Comment:
- As I mentioned earlier Route & Link has to be the same. 

your route says this : ` <Route path='/:id' element={ <BeerDetails />}></Route>`
but in beer details you are saying `/beers/id`

- Also your base URL finishes with beers, so when you add id it looks like this :
`beers1234567`. Do not forget the slash in between!